tests/components/fish_audio/conftest.py

```diff
@@ -0,0 +1,151 @@
+        ) as mock_client,
+        patch(
+            "homeassistant.components.fish_audio.Session",
+            new=mock_client,
+        ),
+    ):
+        client = mock_client.return_value
+        client.get_api_credit.return_value = APICreditEntity(
+            _id="test_id",
+            user_id="test_user",
+            credit=Decimal("100.0"),
+            created_at="2023-01-01T00:00:00Z",
+            updated_at="2023-01-01T00:00:00Z",
+        )
+        yield mock_client
+
+
+@pytest.fixture
+def mock_tts_subentry() -> MagicMock:
+    """Mock a TTS sub-entry."""
+    mock_subentry = MagicMock(spec=ConfigSubentry)
+    mock_subentry.subentry_id = "test-subentry"
+    mock_subentry.subentry_type = "tts"
+    mock_subentry.title = "Test TTS"
+    mock_subentry.data = {CONF_VOICE_ID: "voice-123", CONF_BACKEND: "s1"}
+    return mock_subentry
+
+
+@pytest.fixture
+def mock_config_entry(mock_tts_subentry: MagicMock) -> MockConfigEntry:
+    """Mock a config entry with a TTS sub-entry."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_API_KEY: "key123"},
+        title="Fish Audio",
+    )
+    entry.subentries = {mock_tts_subentry.subentry_id: mock_tts_subentry}
+    return entry
+
+
+@pytest.fixture
+def mock_async_client_connect_error() -> Generator[MagicMock]:
+    """Override Session client with a connection error."""
+    session_mock = _get_session_mock()
+    session_mock.get_api_credit.side_effect = HttpCodeErr(500, "Server Error")
+    with (
+        patch(
+            "homeassistant.components.fish_audio.Session",
+            return_value=session_mock,
+        ) as mock_session,
+        patch(
+            "homeassistant.components.fish_audio.config_flow.Session",
+            new=mock_session,
+        ),
+    ):
+        yield mock_session
+
+
+@pytest.fixture
+def mock_async_client_generic_error() -> Generator[MagicMock]:
+    """Override Session client with a generic error."""
+    session_mock = _get_session_mock()
+    session_mock.get_api_credit.side_effect = Exception("Generic Error")
+    with (
+        patch(
+            "homeassistant.components.fish_audio.Session",
+            return_value=session_mock,
+        ) as mock_session,
+        patch(
+            "homeassistant.components.fish_audio.config_flow.Session",
+            new=mock_session,
+        ),
+    ):
+        yield mock_session
+
+
+@pytest.fixture
+def mock_async_client_auth_error() -> Generator[MagicMock]:
+    """Override Session client with an authentication error."""
+    session_mock = _get_session_mock()
+    session_mock.get_api_credit.side_effect = HttpCodeErr(401, "Auth Error")
+    with (
+        patch(
+            "homeassistant.components.fish_audio.Session",
+            return_value=session_mock,
+        ) as mock_session,
+        patch(
+            "homeassistant.components.fish_audio.config_flow.Session",
+            new=mock_session,
+        ),
+    ):
+        yield mock_session
```

Comment:
I'd either patch it where you need it, or just import `mock_async_client` and overwrite the part you need and then return that mock again.